docs/low-code-spark/extensibility/gem-ui-componets.md

```diff
@@ -0,0 +1,379 @@
+                    )
+                ),
+
+```
+
+</TabItem>
+<TabItem value="scala" label="Scala">
+
+```scala
+
+  StackItem(grow = Some(1)).addElement(
+                  FieldPicker(height = Some("100%"))
+                    .addField(Checkbox("Merge schema"), "mergeSchema")
+                    .addField(
+                      SelectBox("Datetime Rebase Mode")
+                        .addOption("EXCEPTION", "EXCEPTION")
+                        .addOption("CORRECTED", "CORRECTED")
+                        .addOption("LEGACY", "LEGACY"),
+                      "datetimeRebaseMode"
+                    )
+                    .addField(
+                      SelectBox("Int96 Rebase Mode")
+                        .addOption("EXCEPTION", "EXCEPTION")
+                        .addOption("CORRECTED", "CORRECTED")
+                        .addOption("LEGACY", "LEGACY"),
+                      "int96RebaseMode"
+                    )
+                    .addField(Checkbox("Recursive File Lookup"), "recursiveFileLookup")
+                    .addField(TextBox("Path Global Filter").bindPlaceholder(""), "pathGlobFilter")
+                    .addField(TextBox("Modified Before").bindPlaceholder(""), "modifiedBefore")
+                    .addField(TextBox("Modified After").bindPlaceholder(""), "modifiedAfter")
+                )
+
+```
+
+</TabItem>
+</Tabs>
+````
+
+#### Tabs
+
+This component is used to render Tabs in UI like this:
+
+![Tabs](img/tabs.png)
+Example code:
+
+````mdx-code-block
+
+<Tabs>
+
+<TabItem value="py" label="Python">
+
+```py
+                Tabs()
+                    .bindProperty("activeTab")
+                    .addTabPane(
+                    TabPane("Group By", "groupBy").addElement(
+                        ExpTable("Group By Columns").bindProperty("groupBy")
+                    )
+                )
+```
+
+</TabItem>
+<TabItem value="scala" label="Scala">
+
+```scala
+
+ Tabs()
+              .bindProperty("activeTab")
+              .addTabPane(
+                TabPane("Group By", "groupBy")
+                  .addElement(
+                    ExpTable("Group By Columns")
+                      .bindProperty("groupBy")
+                  )
+              )
+```
+
+</TabItem>
+</Tabs>
+````
+
+#### Buttons
+
+Used to add a button on UI.
+
+````mdx-code-block
+
+<Tabs>
+
+<TabItem value="py" label="Python">
+
+```py
+                 Button("Add Transformation").bindOnClick(self.onClickFunc).addElement(
+                            NativeText("Add Transformation")))
+```
+
+</TabItem>
+<TabItem value="scala" label="Scala">
+
+```scala
+
+ Button(
+"Create Dataset",
+children = List(NativeText("Create Dataset")),
+variant = Some(ButtonVariant.tertiaryGrey),
+block = Some(true),
+style = Some(Map("width" -> "100%", "height" -> "48px")),
+onClick = Some { state: Any ⇒
+openCreateDatasetDialog(state.asInstanceOf[WorkflowProcess])
+}
+)
+```
+
+</TabItem>
+</Tabs>
+````
+
+### Forms and Capturing Data
```

Comment:
We can't leave unfinished sections like this

Reply:
Done.